addons/account/tests/test_account_move_entry.py

```diff
@@ -485,3 +485,195 @@ def test_misc_prevent_unlink_posted_items(self):
+            ],
+        })
+        move = self.env['account.move'].create({
+            'type': 'entry',
+            'line_ids': [
+                (0, 0, {
+                    'account_id': self.company_data['default_account_revenue'].id,
+                    'debit': 0.0,
+                    'credit': 100.0,
+                    'tax_ids': [(4, tax_cash_basis.id)],
+                }),
+                (0, 0, {
+                    'name': 'tax line',
+                    'account_id': tax_final_account.id,
+                    'debit': 0.0,
+                    'credit': 20.0,
+                    'tax_repartition_line_id': tax_cash_basis.invoice_repartition_line_ids.filtered(lambda x: x.repartition_type == 'tax').id,
+                    'tax_exigible': False,
+                    'tax_base_amount': 100.00,
+                }),
+                (0, 0, {
+                    'name': 'counterpart lines',
+                    'account_id': self.company_data['default_account_expense'].id,
+                    'debit': 120.0,
+                    'credit': 0.0,
+                }),
+            ],
+        })
+        reverse = move._reverse_moves()
+        # reverse tax account must stay the same as move tax account for caba entries
+        self.assertEqual(
+            move.line_ids.filtered(lambda x: x.tax_line_id).account_id,
+            reverse.line_ids.filtered(lambda x: x.tax_line_id).account_id,
+        )
+
+    def test_invoice_like_entry_reverse_tags(self):
+        """ Reverse entries of entries with invoice like structure (balance, tax), must have it's
+        tax tags set accordingly to repartition lines same as invoices.
+        """
+        tags = self.env['account.account.tag'].create([{
+            'name': "Test tag %s" % i,
+            'applicability': 'taxes',
+            'country_id': self.env.ref('base.us').id,
+        } for i in range(8)])
+        taxes = self.env['account.tax'].create([{
+            'name': "Test tax %s" % i,
+            'amount': 10.0,
+            'invoice_repartition_line_ids': [
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'base',
+                    'tag_ids': [(6, 0, tags[(i * 4)].ids)],
+                }),
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'tax',
+                    'tag_ids': [(6, 0, tags[(i * 4) + 1].ids)],
+                }),
+            ],
+            'refund_repartition_line_ids': [
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'base',
+                    'tag_ids': [(6, 0, tags[(i * 4) + 2].ids)],
+                }),
+                (0, 0, {
+                    'factor_percent': 100,
+                    'repartition_type': 'tax',
+                    'tag_ids': [(6, 0, tags[(i * 4) + 3].ids)],
+                }),
+            ],
+        } for i in range(2)])
```

Comment:
Playing with indices like that does not really help readability ... Maybe add comment explaining which tags go where ? Or just make a helper to create your taxes and create them separately ? (i'd prefer this, personally)